src/sentry/search/events/builder/metrics.py

```diff
@@ -154,21 +153,16 @@ def _get_on_demand_metric_spec(self, field: str) -> Optional[OnDemandMetricSpec]
                     "Must include on demand metrics type when querying on demand"
                 )
 
-            # This feature flag is used to control the rollout of the new environment logic which fixes the previous
-            # implementation. The usage of this flag should be that it is set to true only when the extraction of the
-            # new environment specs has been running for at least 14 days.
-            use_updated_env_logic = features.has(
-                "organizations:on-demand-query-with-new-env-logic",
-                Organization.objects.get_from_cache(id=self.organization_id),
+            spec_version = OnDemandMetricSpecVersioning.get_query_spec_version(
+                self.organization_id, "organizations:on-demand-query-with-new-env-logic"
```

Comment:
@iambriccardo does this look good to you?

Reply:
My idea was to have a central system that given an org id it will check the feature flags on its own internally. Passing a feature flag already implies that the query builder still has knowledge of the feature flag, which it should not.